### PR TITLE
Generate java_plugin targets for annotation processors

### PR DESCRIPTION
**Pull request by illicitonion**

### CODE REVIEW

File: MODULE.bazel [hunk 1]
```diff
@@ -1,8 +1,8 @@
 module(
     name = "rules_jvm_external",
+    version = "6.1",
```
Comment (jin) [inline MODULE.bazel:3]: I feel like I've seen this line being rearranged a ton.. 🤔

File: private/dependency_tree_parser.bzl [hunk 2]
```diff
@@ -319,6 +319,21 @@ genrule(
...
 
+    for annotation_processor in artifact.get("annotation_processors", []):
+        to_return.append(
+            """java_plugin(
+\tname = "{name}",
+\tdeps = [":{jar_target}"],
+\tgenerates_api = True,
+processor_class = "{processor_class}",
+{alias_visibility})""".format(
+                name = "{}__java_plugin__{}".format(target_label, escape(annotation_processor)),
```
Comment (jin) [inline private/dependency_tree_parser.bzl:330]: Can you also provide a `java_plugin()` macro that hides the naming implementation detail in BUILD files?

Something like the `artifact()` macro:

https://github.com/bazelbuild/rules_jvm_external/blob/653bc80e45d317fce7492c90be0acf4c98da8990/private/rules/artifact.bzl#L4

**Review by jin — APPROVED** [review]

File: private/dependency_tree_parser.bzl (hunk 2 above)
Comment (illicitonion) [reply private/dependency_tree_parser.bzl:330]: Nice idea! Done!

File: MODULE.bazel (hunk 1 above)
Comment (illicitonion) [reply MODULE.bazel]: Sigh, reverted the spurious formatting changes :)

**Review by jin — APPROVED** [review]
Mind also updating the README like https://github.com/bazelbuild/rules_jvm_external?tab=readme-ov-file#artifact-helper-macro?

**Review by jin — APPROVED** [review]
thank you :)

**Comment by rdesgroppes** [comment]
@illicitonion, may I request your review on a follow-up PR: #1142?

**Pull request merged**